contracts/zero-ex/src/signature_utils.ts

```diff
@@ -90,24 +108,33 @@ export function ecSignHashWithKey(hash: string, key: string): ECSignature {
     };
 }
 
-function assertSignatureType(signature: Signature, expectedType: SignatureType): void {
-    if (signature.signatureType !== expectedType) {
-        throw new Error(`Expected signature type to be ${expectedType} but received ${signature.signatureType}.`);
+// Parse a hex signature returned by an RPC call into an `ECSignature`.
+function parseRpcSignature(rpcSig: string): ECSignature {
+    if (hexUtils.size(rpcSig) !== 65) {
+        throw new Error(`Invalid RPC signature length: "${rpcSig}"`);
     }
-}
-
-function parsePackedSignatureBytes(signatureBytes: string): Signature {
-    if (hexUtils.size(signatureBytes) !== 66) {
-        throw new Error(`Expected packed signatureBytes to be 66 bytes long: ${signatureBytes}`);
+    // Some providers encode V as 0,1 instead of 27,28.
+    const VALID_V_VALUES = [0, 1, 27, 28];
+    // Some providers return the signature packed as V,R,S and others R,S,V.
+    // Try to guess which encoding it is (with a slight preference for R,S,V).
+    let v = parseInt(rpcSig.slice(-2), 16);
+    if (VALID_V_VALUES.includes(v)) {
+        // Format is R,S,V
+        v = v >= 27 ? v : v + 27;
+        return {
+            r: hexUtils.slice(rpcSig, 0, 32),
+            s: hexUtils.slice(rpcSig, 32, 64),
+            v,
+        };
     }
-    const typeId = parseInt(signatureBytes.slice(-2), 16) as SignatureType;
-    if (!Object.values(SignatureType).includes(typeId)) {
-        throw new Error(`Invalid signatureBytes type ID detected: ${typeId}`);
+    // Format should be V,R,S
```

Comment:
I think you're missing a `v = parseInt(rpcSig.slice(2, 4), 16);` here

Reply:
yikes